src/OpenLoco/Windows/Cheats.cpp

```diff
@@ -240,33 +348,82 @@ namespace OpenLoco::Ui::Windows::Cheats
                     GameCommands::do_81(CheatCommand::clearLoan);
                     WindowManager::invalidateWidget(self->type, self->number, Widx::loan_value);
                     break;
+
+                case Widx::date_change_apply:
+                {
+                    GameCommands::do_81(CheatCommand::modifyDate, _date.year, (int32_t)_date.month, _date.day);
+                    WindowManager::invalidate(WindowType::timeToolbar);
+                    break;
+                }
             }
         }
 
         static void onMouseDown(Window* self, WidgetIndex_t widgetIndex)
         {
             static loco_global<uint16_t, 0x00523376> _clickRepeatTicks;
 
-            currency32_t stepSize{};
+            currency32_t cashStepSize{};
+            int32_t timeStepSize{};
+
             if (*_clickRepeatTicks < 100)
-                stepSize = 1000;
+            {
+                cashStepSize = 1'000;
+                timeStepSize = 1;
+            }
             else if (*_clickRepeatTicks < 200)
-                stepSize = 10000;
+            {
+                cashStepSize = 10'000;
+                timeStepSize = 10;
+            }
+            else if (*_clickRepeatTicks < 300)
+            {
+                cashStepSize = 100'000;
+                timeStepSize = 100;
+            }
             else
-                stepSize = 100000;
+            {
+                cashStepSize = 1'000'000;
+                timeStepSize = 1'000;
+            }
 
             switch (widgetIndex)
             {
                 case Widx::cash_step_decrease:
-                    _cashIncreaseStep = std::max<currency32_t>(_cashIncreaseStep - stepSize, 0);
+                    _cashIncreaseStep = std::max<currency32_t>(_cashIncreaseStep - cashStepSize, 0);
                     WindowManager::invalidateWidget(self->type, self->number, Widx::cash_step_value);
                     break;
 
                 case Widx::cash_step_increase:
-                    _cashIncreaseStep = std::max<currency32_t>(_cashIncreaseStep + stepSize, 0);
+                    _cashIncreaseStep = std::max<currency32_t>(_cashIncreaseStep + cashStepSize, 0);
                     WindowManager::invalidateWidget(self->type, self->number, Widx::cash_step_value);
                     break;
+
+                case Widx::year_step_decrease:
+                    _date.year = std::max<int32_t>(OpenLoco::Scenario::min_year, _date.year - timeStepSize);
```

Comment:
Ive created a Math::Bound::sub and add for this exact situation.

Reply:
Your `sub()` and `add()` functions need a third parameter for the bound for them to be useful here, because they hard code `numeric_limits<T>::min()` (and `max()`), which I only use once here, in `case Widx::year_step_increase:`. I'm happy to add that if you agree

Reply:
Hmm good point. Can you remove the c style casts at least. Modifying the bound function can be done in a follow on